### PR TITLE
[feature][processing] "Array of Offset Features" algorithm

### PR DESCRIPTION
**Pull request by nyalldawson**
This algorithm creates copies of features in a layer, by creating multiple offset versions of the feature. Each copy is displaced by a preset amount in the x/y/z/m axis.

### CODE REVIEW

File: src/analysis/processing/qgsalgorithmarrayfeatures.cpp [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+}
+
+QString QgsArrayFeaturesAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeometry" );
+}
+
+QString QgsArrayFeaturesAlgorithm::outputName() const
+{
+  return QObject::tr( "Offset" );
```
Comment (nirvn) [inline src/analysis/processing/qgsalgorithmarrayfeatures.cpp]: "Offset" is also the output name for the offset line algorithm's output. I'd suggest "Offsets Array", or maybe just add an "s"?

**Comment by nirvn** [comment]
100% cool, 0% practical use of this algorithm:
image

:)

This'll come in handy, esp. with all parameters being dynamic.

**Comment by nirvn** [comment]
Thanks for the name tweaking, looking good.

**Comment by nyalldawson** [comment]
Bonus features: a new "create parallel lines" algorithm, and a port of "offset lines" to c++ (with new dynamic distance parameter)

**Review by alexbruy — APPROVED** [review]

**Pull request merged**